flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/NetworkBufferPoolTest.java

```diff
@@ -518,4 +528,176 @@ public void go() throws Exception {
 			globalPool.destroy();
 		}
 	}
+
+	/**
+	 * Tests {@link NetworkBufferPool#isAvailable()}, verifying that the buffer availability is
+	 * correctly maintained and the future callback is correctly processed.
+	 */
+	@Test
+	public void testBufferAvailabilityAndFutureCallback() throws Exception {
+		final int numBuffers = 10;
+		final int numberOfSegmentsToRequest = 5;
+		final Duration requestSegmentsTimeout = Duration.ofSeconds(10L);
+
+		final NetworkBufferPool globalPool = new NetworkBufferPool(
+				numBuffers,
+				128,
+				numberOfSegmentsToRequest,
+				requestSegmentsTimeout);
+
+		try {
+			assertTrue(globalPool.isAvailable().isDone());
+
+			List<MemorySegment> segments = new ArrayList<>(numberOfSegmentsToRequest);
+			for (int i = 0; i < numberOfSegmentsToRequest; ++i) {
+				MemorySegment segment = globalPool.requestMemorySegment();
+				assertNotNull(segment);
+				segments.add(segment);
+				assertTrue(globalPool.isAvailable().isDone());
+			}
```

Comment:
It is better to extract a separate test for only verifying the logic of `globalPool.requestMemorySegment()`. E.g. before requesting the single segment the global pool is available, and after requesting it becomes unavailable.

This test mixes many functions to be  verified together, then it is difficult to trace the logic and maintain the code future.